### ISSUE TITLE
Lag spikes when breaking/placing blocks, potentially related to chunk multidraw

### ISSUE THREAD

**Issue by Quinteger** [Someone-Else-Was-Taken/Periodic-Table-Reforged#79]
Modpack: Valhelsia v3.4.5

Occasionally, breaking or placing blocks creates a noticeable lag spike. This seems to start occuring after spending some time underground and mining some blocks, not present when you just load the world.
I ran `sparkc` profiling for >90ms ticks and placed/broke some blocks. Here are the results: https://spark.lucko.me/Pkr3Z0e9A8
I then disabled chunk multidraw, the lag was still visibly present, so I wanted to run `sparkc` again. However, when I opened chat and pressed up arrow to get to the stop command, my client crashed with another possibler rendering-related stacktrace.

**Comment by Quinteger** [comment]
Update: I caught this issue again and managed to create `sparkc` profiles both with and without multidraw enabled.

With multidraw: https://spark.lucko.me/FnzhJcamef
Without multidraw: https://spark.lucko.me/7RBvkIs1j2

**Comment by viceversagames** [comment]
Yes, pretty sure this occurs because Sodium needs to update the chunk mesh whenever a block is changed, or the render will be stale. This happens with and without multidraw (no reason to have it off). 

The lag is happening off-thread while the render thread waits for the result (run with `--thread *`), so the root problem may be there (not sure why it spikes, I assume you're in a modpack as I've gotten worse performance than vanilla), but as a temporary "fix" I added a mixin where it wouldn't update if there was another update done in the last 20ms, which only seemed to improve performance a bit around large laggy farms.

That crash is strange, I've gotten the same one with Emojiful, a null check on the IReorderingProcessor resolved it. Not sure what's sending null args, but it's likely not Sodium from what I can tell.

**Comment by Kaleidio** [comment]
you should probably tweak the mixin to not update ever 50ms instead since block update ticks only happen that often, but that's just my guess.
